credentials/tls/certprovider/store.go

```diff
@@ -75,6 +93,25 @@ func (wp *wrappedProvider) Close() {
 	}
 }
 
+// Close overrides the Close method of the embedded provider to avoid release the
+// already released reference.
+func (w *singleCloseWrappedProvider) Close() {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	w.provider.Close()
```

Comment:
Can we do this call without the lock held?

Reply:
I have replaced the use of `sync.RWMutex` with `atomic.Pointer`